kernel/src/syscall/arch/generic.rs

```diff
@@ -0,0 +1,404 @@
+            semget::sys_semget,
+            semop::{sys_semop, sys_semtimedop},
+            sendfile::sys_sendfile,
+            sendmmsg::sys_sendmmsg,
+            sendmsg::sys_sendmsg,
+            sendto::sys_sendto,
+            set_ioprio::sys_ioprio_set,
+            set_priority::sys_set_priority,
+            set_robust_list::sys_set_robust_list,
+            set_tid_address::sys_set_tid_address,
+            setdomainname::sys_setdomainname,
+            setfsgid::sys_setfsgid,
+            setfsuid::sys_setfsuid,
+            setgid::sys_setgid,
+            setgroups::sys_setgroups,
+            sethostname::sys_sethostname,
+            setitimer::{sys_getitimer, sys_setitimer},
+            setns::sys_setns,
+            setpgid::sys_setpgid,
+            setregid::sys_setregid,
+            setresgid::sys_setresgid,
+            setresuid::sys_setresuid,
+            setreuid::sys_setreuid,
+            setsid::sys_setsid,
+            setsockopt::sys_setsockopt,
+            setuid::sys_setuid,
+            setxattr::{sys_fsetxattr, sys_lsetxattr, sys_setxattr},
+            shutdown::sys_shutdown,
+            sigaltstack::sys_sigaltstack,
+            signalfd::sys_signalfd4,
+            socket::sys_socket,
+            socketpair::sys_socketpair,
+            stat::{sys_fstat, sys_fstatat},
+            statfs::{sys_fstatfs, sys_statfs},
+            statx::sys_statx,
+            symlink::sys_symlinkat,
+            sync::{sys_sync, sys_syncfs},
+            sysinfo::sys_sysinfo,
+            tgkill::sys_tgkill,
+            timer_create::{sys_timer_create, sys_timer_delete},
+            timer_settime::{sys_timer_gettime, sys_timer_settime},
+            timerfd_create::sys_timerfd_create,
+            timerfd_gettime::sys_timerfd_gettime,
+            timerfd_settime::sys_timerfd_settime,
+            truncate::{sys_ftruncate, sys_truncate},
+            umask::sys_umask,
+            umount::sys_umount,
+            uname::sys_uname,
+            unlink::sys_unlinkat,
+            unshare::sys_unshare,
+            utimens::sys_utimensat,
+            wait4::sys_wait4,
+            waitid::sys_waitid,
+            write::sys_write,
+        };
+    };
+}
+
+pub(super) use import_generic_syscall_entries;
+
+/// Defines syscalls by combining architecture-specific definitions with the generic syscall table.
+///
+/// Generic syscalls refer to syscalls whose numbers are standardized across multiple architectures.
+/// By using this macro, you only need to define the arch-specific syscalls;
+/// the generic ones are integrated automatically.
+///
+/// Reference: https://elixir.bootlin.com/linux/v6.18/source/include/uapi/asm-generic/unistd.h.
+///
+/// # Example
+///
+/// Defining architecture-specific syscalls:
+///
+/// ```rust
+/// define_syscalls_with_generic_syscall_table! {
+///     SYS_EXAMPLE1 = 500 => sys_example1(args[..0]);
+///     SYS_EXAMPLE2 = 501 => sys_example2(args[..1], &user_ctx);
+/// }
+/// ```
```

Comment:
```suggestion
/// Reference: <https://elixir.bootlin.com/linux/v6.18/source/include/uapi/asm-generic/unistd.h>.
///
/// # Examples
///
/// Define architecture-specific syscalls `example1` and `example2` with generic syscalls:
///
/// ```rust
/// define_syscalls_with_generic_syscall_table! {
///     SYS_EXAMPLE1 = 500 => sys_example1(args[..0]);
///     SYS_EXAMPLE2 = 501 => sys_example2(args[..1], &user_ctx);
/// }
/// ```
```